diff --git a/scripts/react_native_pods.rb b/scripts/react_native_pods.rb
index 3d86c4f3de4e3db91dee20085545aacacecebf5b..28abb4df3c5e0028d341201cae87dae1b966ad43 100644
--- a/scripts/react_native_pods.rb
+++ b/scripts/react_native_pods.rb
@@ -34,7 +34,7 @@ def use_react_native! (options={})
   new_arch_enabled = ENV['RCT_NEW_ARCH_ENABLED'] == '1'
 
   # Include DevSupport dependency
-  production = options[:production] ||= false
+  production = ENV['PRODUCTION'] == '1'
 
   # Include Hermes dependencies
   hermes_enabled = options[:hermes_enabled] != nil ? options[:hermes_enabled] : true
